`TestService_ProxySSH_Errors` flakiness

## Failure

#### Link(s) to logs

* https://github.com/gravitational/teleport/actions/runs/4316290164/jobs/7531867607
* https://github.com/gravitational/teleport/actions/runs/4323993127/jobs/7548250628

#### Relevant snippet

```
FAIL: github.com/gravitational/teleport/lib/srv/transport/transportv1
FAIL: github.com/gravitational/teleport/lib/srv/transport/transportv1.TestService_ProxySSH_Errors/no_access_checker_terminates_stream
FAIL: github.com/gravitational/teleport/lib/srv/transport/transportv1.TestService_ProxySSH_Errors
===================================================
OUTPUT github.com/gravitational/teleport/lib/srv/transport/transportv1.TestService_ProxySSH_Errors
===================================================
=== RUN   TestService_ProxySSH_Errors
=== PAUSE TestService_ProxySSH_Errors
=== CONT  TestService_ProxySSH_Errors
--- FAIL: TestService_ProxySSH_Errors (0.02s)
===================================================
OUTPUT github.com/gravitational/teleport/lib/srv/transport/transportv1.TestService_ProxySSH_Errors/no_access_checker_terminates_stream
===================================================
=== RUN   TestService_ProxySSH_Errors/no_access_checker_terminates_stream
=== PAUSE TestService_ProxySSH_Errors/no_access_checker_terminates_stream
make: *** [test-go-unit] Error 1
=== CONT  TestService_ProxySSH_Errors/no_access_checker_terminates_stream
    transport_test.go:343: 
        	Error Trace:	/__w/teleport/teleport/lib/srv/transport/transportv1/transport_test.go:343
        	            				/__w/teleport/teleport/lib/srv/transport/transportv1/transport_test.go:419
        	Error:      	Received unexpected error:
        	            	EOF
        	Test:       	TestService_ProxySSH_Errors/no_access_checker_terminates_stream
--- FAIL: TestService_ProxySSH_Errors/no_access_checker_terminates_stream (0.04s)
ERRO [TRANSPORT] ssh stream terminated unexpectedly: rpc error: code = Canceled desc = context canceled transportv1/transport.go:218
===================================================
Makefile:593: recipe for target 'test-go-unit' failed
```

--- a/lib/srv/transport/transportv1/transport_test.go
+++ b/lib/srv/transport/transportv1/transport_test.go
@@ -367,8 +367,20 @@ func TestService_ProxySSH_Errors(t *testing.T) {
 				}}))
 
 				resp, err := stream.Recv()
-				require.ErrorIs(t, err, trace.AccessDenied("no access checker"))
 				require.Nil(t, resp)
+				switch {
+				// The server will attempt to get the authz context prior to receiving the first
+				// message from the client which may terminate the stream and result in an EOF.
+				case errors.Is(err, io.EOF):
+				// The client send may be completed prior to the server getting the authz context
+				// which will result in the client actually receiving the error from getting the
+				// authz context instead of an EOF.
+				case errors.Is(err, trace.AccessDenied("no access checker")):
+				// All other errors indicate that something went wrong
+				default:
+					t.Fatalf("expected either EOF or Access Denied, got %v", err)
+				}
+
 			},
 		},
 		{

